File: Easy/5. Non Constructible Sum/nonConstructibleChange.py
def nonConstructibleChange(coins):
	currentMaxChange = 0
	coins.sort()
	for i in range(len(coins)):
		if currentMaxChange + 1 < coins[i]:
			return currentMaxChange + 1
		currentMaxChange += coins[i]
	return currentMaxChange + 1

if __name__ == "__main__":
	coins = [5, 7, 1, 1, 2, 3, 22]
	print(f"Minimum amount of change that cannot be created: {nonConstructibleChange(coins)}")

	if 0:
		print("hi")

File: Easy/5. Non Constructible Sum/nonConstructibleChange.go
package main

import (
	"fmt"
	"sort"
)

func NonConstructibleChange(coins []int) int {
	sort.Ints(coins)
	currentMaxChange := 0

	for _, coin := range coins {
		if coin > currentMaxChange+1 {
			return currentMaxChange + 1
		}
		currentMaxChange += coin
	}
	return currentMaxChange + 1
}

func main() {
	coins := []int{5, 7, 1, 1, 2, 3, 22}
	fmt.Println("Minimum amount of change that cannot be created: ", NonConstructibleChange(coins))
}
